pkg/resourcehandlers/github/github_resource_handler.go

```diff
@@ -335,24 +334,53 @@ func (gh *GitHub) Accept(uri string) bool {
 
 // ResolveNodeSelector recursively adds nodes built from tree entries to node
 // ResolveNodeSelector implements resourcehandlers/ResourceHandler#ResolveNodeSelector
-func (gh *GitHub) ResolveNodeSelector(ctx context.Context, node *api.Node, excludePaths []string, frontMatter map[string]interface{}, excludeFrontMatter map[string]interface{}, depth int32) error {
-	var (
-		rl  *ResourceLocator
-		err error
-	)
-	if rl, err = gh.URLToGitHubLocator(ctx, node.NodeSelector.Path, true); err != nil {
-		return err
+func (gh *GitHub) ResolveNodeSelector(ctx context.Context, node *api.Node, excludePaths []string, frontMatter map[string]interface{}, excludeFrontMatter map[string]interface{}, depth int32) ([]*api.Node, error) {
+	rl, err := gh.URLToGitHubLocator(ctx, node.NodeSelector.Path, true)
+	if err != nil {
+		return nil, err
 	}
-	if rl != nil {
-		// build node subnodes hierarchy from cache (URLToGitHubLocator populates the cache)
-		childResourceLocators := gh.cache.GetSubset(rl.String())
-		if err = buildNodes(node, excludePaths, frontMatter, excludeFrontMatter, depth, childResourceLocators, gh.cache, 0); err != nil {
-			return err
-		}
-		// finally cleanup folder entries from contentSelectors
-		cleanupNodeTree(node)
+	if rl == nil {
+		return nil, nil
```

Comment:
That's a bit way too silent, isn't it? When can we get `nil` from `URLToGitHubLocator` without an error? And isn't that an indication of a problem? See my comment above that with this you are heading to a NPE in using code. 
Also see this [PR](https://github.com/gardener/docforge/pull/89) that fixes exactly a situation when the method returns were `nil, nil`